File: test-api-login.js
const http = require('http');

const loginData = JSON.stringify({
  email: 'admin@fulexo.com',
  password: 'demo123'
});

const options = {
  hostname: 'localhost',
  port: 3000,
  path: '/api/auth/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': loginData.length
  }
};

console.log('=== LOGIN TEST ===');
console.log('Request:', loginData);

const req = http.request(options, (res) => {
  let data = '';
  
  console.log(`Status Code: ${res.statusCode}`);
  console.log(`Headers:`, JSON.stringify(res.headers, null, 2));
  
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    try {
      const jsonData = JSON.parse(data);
      console.log('\nResponse:', JSON.stringify(jsonData, null, 2));
      
      if (res.statusCode === 200 || res.statusCode === 201) {
        console.log('\n✓ LOGIN BASARILI!');
        const cookies = res.headers['set-cookie'];
        console.log('Cookies:', cookies);
      } else {
        console.log('\n✗ LOGIN BASARISIZ!');
      }
    } catch (e) {
      console.log('Raw Response:', data);
      console.log('Parse Error:', e.message);
    }
  });
});

req.on('error', (error) => {
  console.error('Request Error:', error);
});

req.write(loginData);
req.end();

File: test-api-login.py
#!/usr/bin/env python3
import requests
import json

# Login
login_data = {
    "email": "admin@fulexo.com",
    "password": "demo123"
}

print("=== LOGIN TEST ===")
response = requests.post(
    "http://localhost:3000/api/auth/login",
    json=login_data,
    headers={"Content-Type": "application/json"}
)

print(f"Status Code: {response.status_code}")
print(f"Response: {json.dumps(response.json(), indent=2)}")

if response.status_code == 200 or response.status_code == 201:
    print("\n✓ LOGIN BASARILI!")
    cookies = response.cookies
    print(f"Cookies: {cookies}")
    
    # Test dashboard endpoint
    print("\n=== DASHBOARD TEST ===")
    dash_response = requests.get(
        "http://localhost:3000/api/dashboard/stats",
        cookies=cookies
    )
    print(f"Status: {dash_response.status_code}")
    if dash_response.status_code == 200:
        print(f"Dashboard Data: {json.dumps(dash_response.json(), indent=2)}")
else:
    print("\n✗ LOGIN BASARISIZ!")

